File: 6.Semestr/1.Programming in Python(BI-PYT.21)/2.Semestral work/app/app/schemas/calendar.py
"""DTO schemes for Calendar entity."""
from pydantic import BaseModel


class Rules(BaseModel):
    """Represents rules of user."""
    night_time: bool
    reservation_more_24_hours: bool
    in_advance_hours: int
    in_advance_minutes: int
    in_advance_day: int


class CalendarBase(BaseModel):
    """Shared properties of Calendar."""
    collision_with_itself: bool | None = None
    collision_with_calendar: list[str] | None = None
    club_member_rules: Rules | None = None
    active_member_rules: Rules | None = None
    manager_rules: Rules | None = None
    mini_services: list[str] | None = None


class CalendarCreate(CalendarBase):
    """Properties to receive via API on creation."""
    calendar_id: str
    service_alias: str
    reservation_type: str
    event_name: str
    max_people: int
    collision_with_itself: bool
    club_member_rules: Rules
    active_member_rules: Rules
    manager_rules: Rules


class CalendarUpdate(CalendarBase):
    """Properties to receive via API on update."""
    service_alias: str | None = None
    reservation_type: str | None = None
    event_name: str | None = None
    max_people: int | None = None
    collision_with_itself: bool | None = None
    collision_with_calendar: list[str] | None = None
    club_member_rules: Rules | None = None
    active_member_rules: Rules | None = None
    manager_rules: Rules | None = None
    mini_services: list[str] | None = None


class CalendarInDBBase(CalendarBase):
    """Base model for user in database."""
    calendar_id: str
    service_alias: str
    reservation_type: str
    event_name: str
    max_people: int

    # pylint: disable=too-few-public-methods
    # reason: Config class only needs to set orm_mode to True.
    class Config:
        """Config class for database user model."""
        from_attributes = True


class Calendar(CalendarInDBBase):
    """Additional properties of calendar to return via API."""


class CalendarInDB(CalendarInDBBase):
    """Additional properties stored in DB"""

File: 6.Semestr/2.Programming for the Android operating system(BI-AND.21)/3.Semestral work/BukReservationApp/app/src/main/java/cz/cvut/fit/kuznear1/bukreservationapp/features/calendar/domain/Calendar.kt
package cz.cvut.fit.kuznear1.bukreservationapp.features.calendar.domain

import cz.cvut.fit.kuznear1.bukreservationapp.features.calendar.data.api.ApiCalendar
import kotlinx.serialization.Serializable

@Serializable
data class Calendar(
    val calendar_id: String,
    val service_alias: String,
    val reservation_type: String,
    val event_name: String,
    val max_people: Int,
    val collision_with_itself: Boolean,
    val collision_with_calendar: List<String>,
    val club_member_rules: ApiCalendar.MemberRules,
    val active_member_rules: ApiCalendar.MemberRules,
    val manager_rules: ApiCalendar.MemberRules,
    val mini_services: List<String>?
)

fun CreateDefaultCalendar(): Calendar {
    val rules = ApiCalendar.MemberRules(
        night_time = false,
        reservation_more_24_hours = false,
        in_advance_hours = 0,
        in_advance_minutes = 0,
        in_advance_day = 0,
    )

    return Calendar(
        calendar_id = "",
        service_alias = "",
        reservation_type = "",
        event_name = "",
        max_people = 0,
        collision_with_itself = false,
        collision_with_calendar = emptyList(),
        club_member_rules = rules,
        active_member_rules = rules,
        manager_rules = rules,
        mini_services = null
    )
}